src/posit/connect/resources.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+from typing import Generic, List, Optional, TypeVar
+
+
+T = TypeVar("T")
+
+
+class Resources(ABC, Generic[T]):
+    @abstractmethod
+    def create(self, *args, **kwargs) -> T:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def delete(self, *args, **kwargs) -> None:
```

Comment:
Does `delete()` make sense for these collections? Would this be a bulk delete action?

Reply:
Based on this method definition, the implementation can do either. We haven't tried injecting the session object into the associated Resource yet. This should be solved as part of #23.

Typically, I would expect this method to act on a collection, which would either be the syntax sugar of `user.delete() for user in user` (pseudo-code) or a direct call to a DELETE endpoint that supports deleting multiple items in a collection.